### PR TITLE
Create a Python Lab setup package

### PR DESCRIPTION
**Pull request by molly-moen**
We have so far found 2 packages that we need to monkey patch for use in Python Lab. The first is matplotlib, which we patch to send graphs as a base64 encoded string we can parse on the Python Lab side to display properly. This was originally written as a TypeScript constant appended to all user's code in [this PR](https://github.com/code-dot-org/code-dot-org/pull/57782). 
In order to support fetch requests, we wanted to be able to have the same allow-list of urls that we have for App Lab. This is controlled by the [xhr_proxy_controller](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/xhr_proxy_controller.rb). In order to utilize this, we need to wrap all user's calls to requests with the proxied url. There are multiple methods in `requests` that we would need to patch in order to do that. It was going to be unwieldy and hard to test this if we kept with the constant in TypeScript.

The proposed solution here is to create our own package with a `setup_pythonlab()` method we can prefix to the user's code. This method applies the two patches, and can do more in the future if we would like. 

This package will only work once we have unblocked loading pyodide from code.org rather than the pyodide cdn. This currently works only locally, so I was able to test this work in [molly/block-urls-python-lab](https://github.com/code-dot-org/code-dot-org/compare/staging...molly/block-urls-python-lab). For now I'm opening this PR to get feedback on this approach and get the code ready for when we have fixed the pyodide issue.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
-# python-lab-packages
+# pythonlab-packages
 Packages for use in Python Lab
+
+## Setup
+- Install `pyenv`. 
+    - See instructions [here](https://github.com/pyenv/pyenv?tab=readme-ov-file#getting-pyenv)
+- Install python.
+    ```
+    pyenv install 3.12.2
```
Comment (snickell) [inline README.md]: Since there's already a .python-version file, might consider: ```pyenv install $(cat .python-version)```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,25 @@
-# python-lab-packages
+# pythonlab-packages
 Packages for use in Python Lab
```
Comment (snickell) [inline README.md]: I'd love to see a little summary of why and what this package is patching, very similar to the PR description.

File: packages/pythonlab_setup/pythonlab_setup/patch_matplotlib.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    def show():
+        buf = BytesIO()
+        matplotlib.pyplot.savefig(buf, format='png')
+        buf.seek(0)
+        # encode to a base64 str
+        img = base64.b64encode(buf.read()).decode('utf-8')
+        print(f'{matplotlib_img_tag} {img}')
+        matplotlib.pyplot.clf()
+
+    matplotlib.pyplot.show = show
```
Comment (snickell) [inline packages/pythonlab_setup/pythonlab_setup/patch_matplotlib.py:25]: Maybe a short comment explaining why this `show()` is different than the stock pyplot.show? i.e. "why is this happenning", similar to the README.md.

**Review by snickell — APPROVED** [review]
Looks good, I appreciate the use of modern python packaging idioms. I'd prefer to see a little more explanation in the code for what its doing and why.... as it is now I think a future person would have no idea why we are doing these things unless they found the PR description (which is great!)

File: README.md (hunk 2 above)
Comment (molly-moen) [reply README.md]: Good idea!

**Pull request merged**